packages/cli/lib/LocalDevManager.js

```diff
@@ -0,0 +1,382 @@
+        text: i18n(`${i18nKey}.exitingSucceed`),
+      });
+    } else {
+      this.spinnies.fail('cleanupMessage', {
+        text: i18n(`${i18nKey}.exitingFail`),
+      });
+    }
+
+    process.exit(exitCode);
+  }
+
+  startUploadQueue() {
+    this.uploadQueue.start();
+
+    this.spinnies.removeAll();
+    this.spinnies.add('devModeRunning', {
+      text: i18n(`${i18nKey}.running`, {
+        projectName: this.projectConfig.name,
+        accountIdentifier: uiAccountDescription(this.targetAccountId),
+      }),
+      isParent: true,
+    });
+    this.spinnies.add('devModeStatus', {
+      text: i18n(`${i18nKey}.status.clean`),
+      status: 'non-spinnable',
+      indent: 1,
+    });
+    this.spinnies.add('quitMessage', {
+      text: i18n(`${i18nKey}.quitHelper`),
+      status: 'non-spinnable',
+      indent: 1,
+    });
+  }
+
+  updateDevModeStatus(langKey) {
+    this.spinnies.update('devModeStatus', {
+      text: i18n(`${i18nKey}.status.${langKey}`),
+      status: 'non-spinnable',
+    });
+  }
+
+  async pauseUploadQueue() {
+    this.spinnies.removeAll();
+    this.spinnies.add('uploading', {
+      text: i18n(`${i18nKey}.upload.uploadingChanges`),
+      isParent: true,
+    });
+
+    this.uploadQueue.pause();
+    await this.uploadQueue.onIdle();
+  }
+
+  hasAnyUnsupportedStandbyChanges() {
+    return this.standbyChanges.some(({ supported }) => !supported);
+  }
+
+  async createNewStagingBuild() {
+    try {
+      const { buildId } = await provisionBuild(
+        this.targetAccountId,
+        this.projectConfig.name
+      );
+      this.currentStagedBuildId = buildId;
+    } catch (err) {
+      logger.debug(err);
+      if (isSpecifiedError(err, { subCategory: ERROR_TYPES.PROJECT_LOCKED })) {
+        await cancelStagedBuild(this.targetAccountId, this.projectConfig.name);
+        logger.log(i18n(`${i18nKey}.previousStagingBuildCancelled`));
+      }
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
+
+  async startWatching() {
+    if (!this.preventUploads) {
+      await this.createNewStagingBuild();
+    }
+
+    this.watcher.on('add', async filePath => {
+      this.handleWatchEvent(filePath, WATCH_EVENTS.add);
+    });
+    this.watcher.on('change', async filePath => {
+      this.handleWatchEvent(filePath, WATCH_EVENTS.change);
+    });
+    this.watcher.on('unlink', async filePath => {
+      this.handleWatchEvent(filePath, WATCH_EVENTS.unlink);
+    });
+    this.watcher.on('unlinkDir', async filePath => {
+      this.handleWatchEvent(filePath, WATCH_EVENTS.unlinkDir);
+    });
+  }
+
+  async handleWatchEvent(filePath, event) {
+    const changeInfo = {
+      event,
+      filePath,
+      remotePath: path.relative(this.projectSourceDir, filePath),
+    };
+
+    const isSupportedChange = await this.notifyServers(changeInfo);
+
+    if (isSupportedChange) {
+      this.addChangeToStandbyQueue({ ...changeInfo, supported: true });
+      return;
+    } else if (this.preventUploads) {
```

Comment:
Should this check for `preventUploads` first?

Reply:
Good call. I'll fix this

Reply:
done in [a4151f7](https://github.com/HubSpot/hubspot-cli/pull/830/commits/a4151f731e6d699d8edd834e2142bdc77d7f0add)